runtime/common/src/try_runtime.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! common try-runtime only tests for runtimes.
+
+/// register all inactive nominators for fast-unstake, and progress until they have all been
+/// processed.
+pub fn migrate_all_inactive_nominators<T: pallet_fast_unstake::Config + pallet_staking::Config>()
+where
+	<T as frame_system::Config>::RuntimeEvent: TryInto<pallet_fast_unstake::Event<T>>,
+{
+	use frame_support::{
+		dispatch::RawOrigin,
+		traits::{Get, Hooks},
+	};
+	use pallet_fast_unstake::{Pallet as FastUnstake, *};
+	use pallet_staking::*;
+	use sp_std::{collections::btree_set::BTreeSet, prelude::*};
+
+	let mut unstaked_ok = 0;
+	let mut unstaked_err = 0;
+	let mut unstaked_slashed = 0;
+
+	let all_stakers = Ledger::<T>::iter().map(|(ctrl, l)| (ctrl, l.stash)).collect::<BTreeSet<_>>();
+	let mut all_exposed = BTreeSet::new();
+	ErasStakers::<T>::iter().for_each(|(_, val, expo)| {
+		all_exposed.insert(val);
+		expo.others.iter().for_each(|ie| {
+			all_exposed.insert(ie.who.clone());
+		})
+	});
+
+	let eligible = all_stakers
+		.iter()
+		.filter_map(|(ctrl, stash)| if all_exposed.contains(stash) { None } else { Some(ctrl) })
+		.collect::<Vec<_>>();
+
+	log::info!(
+		target: "runtime::test",
+		"registering {} out of {} stakers for fast-unstake",
+		eligible.len(),
+		all_stakers.len()
+	);
+	for ctrl in eligible {
+		if let Err(why) =
+			FastUnstake::<T>::register_fast_unstake(RawOrigin::Signed(ctrl.clone()).into())
+		{
+			log::warn!(target: "runtime::test", "failed to register {:?} due to {:?}", ctrl, why);
+		}
+	}
+
+	log::info!(target: "runtime::test", "registered {} successfully, starting at {:?}.", Queue::<T>::count(), frame_system::Pallet::<T>::block_number());
+	while Queue::<T>::count() != 0 || Head::<T>::get().is_some() {
+		// assume half of the block weight is available.
+		let now = frame_system::Pallet::<T>::block_number();
+		let weight = <T as frame_system::Config>::BlockWeights::get().max_block;
+		let consumed = FastUnstake::<T>::on_idle(now, weight);
+		log::debug!(target: "runtime::test", "consumed {:?} ({})", consumed, consumed.ref_time() as f32 / weight.ref_time() as f32);
+
+		frame_system::Pallet::<T>::read_events_no_consensus()
```

Comment:
This is only in consensus code since its a migration, or?  
Maybe put a comment so that no one copy&pastes this :see_no_evil: